packages/legend-query/src/components/QueryBuilderResultPanel.tsx

```diff
@@ -39,17 +42,244 @@ import {
+            operators
+              .map((op) => op.getLabel())
+              .includes(
+                (
+                  n as QueryBuilderPostFilterTreeConditionNodeData
+                ).condition.operator.getLabel(),
+              ),
+        );
+
+    const generateNewPostFilterConditionNodeData = async (
+      projectionColumnState: QueryBuilderProjectionColumnState,
+      operator: QueryBuilderPostFilterOperator,
+    ): Promise<void> => {
+      let postFilterConditionState: PostFilterConditionState;
+      try {
+        postFilterConditionState = new PostFilterConditionState(
+          postFilterState,
+          projectionColumnState,
+          undefined,
+          operator,
+        );
+        if (
+          projectionColumnState instanceof
+          QueryBuilderDerivationProjectionColumnState
+        ) {
+          await flowResult(
+            projectionColumnState.fetchDerivationLambdaReturnType(),
+          );
+        }
+        postFilterConditionState.setValue(
+          postFilterConditionState.operator.getDefaultFilterConditionValue(
+            postFilterConditionState,
+          ),
+        );
+        (postFilterConditionState.value as InstanceValue).values[0] =
+          event?.value;
+      } catch (error) {
+        assertErrorThrown(error);
+        applicationStore.notifyWarning(error.message);
+        return;
+      }
+      postFilterState.addNodeFromNode(
+        new QueryBuilderPostFilterTreeConditionNodeData(
+          undefined,
+          postFilterConditionState,
+        ),
+        undefined,
+      );
+    };
+
+    const updateExistingPostFilterConditionNodeData = (
+      existingPostFilterNode: QueryBuilderPostFilterTreeNodeData[],
+      isFilterBy: boolean,
+    ): void => {
+      const conditionState = (
+        existingPostFilterNode[0] as QueryBuilderPostFilterTreeConditionNodeData
+      ).condition;
+      if (
+        conditionState.operator.getLabel() ===
+        (isFilterBy
+          ? postFilterEqualOperator
+          : postFilterNotEqualOperator
+        ).getLabel()
+      ) {
+        if (
+          !(
+            conditionState.value instanceof InstanceValue &&
+            conditionState.value.values.includes(event?.value)
+          )
+        ) {
+          const currentValueSpecificaton = conditionState.value;
+          const newValueSpecification =
+            conditionState.operator.getDefaultFilterConditionValue(
+              conditionState,
+            );
+          instanceValue_changeValue(
+            newValueSpecification as InstanceValue,
+            event?.value,
+            0,
+          );
+          conditionState.changeOperator(
+            isFilterBy ? postFilterInOperator : postFilterNotInOperator,
+          );
+          (conditionState.value as InstanceValue).values = [
+            currentValueSpecificaton,
+            newValueSpecification,
+          ];
+        }
+      } else {
+        if (
+          !(
```

Comment:
when do you encounter this case ?

Reply:
existing post filter: id is 1, user action: filter by id = 1 
existing post filter:  id is in [1 ,2 ], user action : filter by id = 1

Reply:
as discussed, lets cleanup the if statements to make it easier to read.